### ISSUE TITLE
Editor: Implement a mouse pointer which just triggers a click event but does nothing else on left click

### ISSUE THREAD

**Issue by bunnybot** [widelands/widelands#6913]
[<img align='left' src='https://codeberg.org/avatars/94062fb4c3e503b6434a88938aca6486630fcdfe4d13f557dc2ef771ba3943d8' alt='frankystone' height='48' />](https://codeberg.org/frankystone)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/5273)*
*Created on Thu Nov 13 13:00:58 CET 2025 by [**frankystone**](https://codeberg.org/frankystone)*

---

### Describe the Problem

If the editor starts the info tool is activated as default. This can be very annoying if you are used to e.g. scroll the view with left click. Each click at editor startup opens a new info window, which is at this time pretty useless.

Handling (left) click events without an activated tool will be also needed for #6703 .

### Solution

Implement something like a "click-only-tool" which does not change the map, but triggers a map click event. Activate this new tool at start of the editor so a click on the map just does nothing. 

It should also be possible to activate the new click-only-tool with lua to make it possible to activate it when #6703 is merged and one wants to use  `subscribe_to_field_clicked()`.

### Additional Context

Probably related:

*  #5588
*  #1397

**Comment by bunnybot** [comment]
[<img align='left' src='https://codeberg.org/avatars/593dc43e2444c25db9481960e5b693ab' alt='tothxa' height='48' />](https://codeberg.org/tothxa)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/5273#issuecomment-8266107)*
*On Thu Nov 13 18:33:49 CET 2025, [**Tóth András (tothxa)**](https://codeberg.org/tothxa) wrote:*

---

I think there are 2 separate issues here:

1. A safe "do nothing" tool against accidental clicks
2. A custom toolbox where UI add-ons can register lua tools. This should be properly handled by the C++ code, so that you only have to provide a lua function name (that receives the coordinates, the button, and modifier keys as arguments), an icon image, and a textual name. It shouldn't need messing with subscribers and timers.

**Comment by bunnybot** [comment]
[<img align='left' src='https://codeberg.org/avatars/a4ad2f9a2504f052397d8ba82ed48ba1' alt='Nordfriese' height='48' />](https://codeberg.org/Nordfriese)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/5273#issuecomment-8308223)*
*On Sun Nov 16 20:09:47 CET 2025, [**Benedikt Straub (Nordfriese)**](https://codeberg.org/Nordfriese) wrote:*

---

+1 for such a tool, but it should not be activateable by the user IMHO since there's very little use for it really. Only scripts should receive the ability to activate this editor tool, so custom tooling can work well.
Relatedly also in my to-do list is the possibility for plugins to hook into the editor Undo/Redo stack. All these things together would allow fully-fledged custom editor tools. I think additionally providing a "toolbox toolkit" would overcomplicate things TBH, the editor tool system is nastily complex enough already…

**Comment by bunnybot** [comment]
[<img align='left' src='https://codeberg.org/avatars/94062fb4c3e503b6434a88938aca6486630fcdfe4d13f557dc2ef771ba3943d8' alt='frankystone' height='48' />](https://codeberg.org/frankystone)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/5273#issuecomment-8308622)*
*On Sun Nov 16 20:53:46 CET 2025, [**frankystone**](https://codeberg.org/frankystone) wrote:*

---

This was not meant to be activate-able by the user, more for situations where no other tool is activated. e.g:

* at editor startup
* if no tool window is open, or all tool windows are closed like in #1397

Regarding a "toolbox toolkit" i have no meaning. But for #6703 it would be very welcome if the scripts wont need to implement the loops themselves.

**Comment by bunnybot** [comment]
[<img align='left' src='https://codeberg.org/avatars/593dc43e2444c25db9481960e5b693ab' alt='tothxa' height='48' />](https://codeberg.org/tothxa)*[Mirrored from Codeberg](https://codeberg.org/wl/widelands/issues/5273#issuecomment-8359317)*
*On Thu Nov 20 01:34:48 CET 2025, [**Tóth András (tothxa)**](https://codeberg.org/tothxa) wrote:*

---

After looking at the implementation of notifications for reviewing #6703, I can see that the minimal callback functions are much safer than calling directly into lua. But allowing to handle `field_clicked` both by the normal code and add-ons/scripts is not right. Like this feature request shows too.

**Comment by bunnybot** [comment]
Assigned to [<img align='left' src='https://codeberg.org/avatars/a4ad2f9a2504f052397d8ba82ed48ba1' alt='Nordfriese' height='48' />](https://codeberg.org/Nordfriese) [**Benedikt Straub (Nordfriese)**](https://codeberg.org/Nordfriese) *( `@cb/Nordfriese` )*
